<?xml version="1.0"?>
<!-- SESSION ID: 1523559094 -->
<launch>

  <!-- Calibration parameters -->
  <arg name="lock_world_frame" default="false" />
  <group if="$(arg lock_world_frame)">
    <arg name="fixed_sensor_id" />
    <rosparam command="load" file="$(find opt_calibration)/conf/camera_poses.yaml" />
  </group>

  <!-- Network parameters -->
  <arg name="num_sensors"   default="2" />

  <arg name="sensor_0_id"   default="kinect_01" />
  <arg name="sensor_0_name" default="$(arg sensor_0_id)" />

  <arg name="sensor_1_id"   default="kinect_02" />
  <arg name="sensor_1_name" default="$(arg sensor_1_id)" />

  <arg name="rows"          default="6" />
  <arg name="cols"          default="5" />
  <arg name="cell_width"    default="0.12" />
  <arg name="cell_height"   default="0.12" />

  <!-- Opening Rviz for visualization -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find opt_calibration)/conf/opt_calibration.rviz" />

  <!-- Plot calibration status -->
  <node name="opt_calibration_status_plot" pkg="opt_calibration" type="status_plot.py" output="screen">
    <remap from="~calibration_status" to="/opt_calibration/status" />
  </node>

  <!-- Launching calibration -->
  <node pkg="opt_calibration" type="opt_calibration" name="opt_calibration" output="screen">

    <param unless="$(arg lock_world_frame)" name="world_computation" value="last_checkerboard" />
    <param     if="$(arg lock_world_frame)" name="world_computation" value="update" />
    <param     if="$(arg lock_world_frame)" name="fixed_sensor/name" value="/$(arg fixed_sensor_id)" />

    <param name="num_sensors"           value="$(arg num_sensors)" />

    <param name="rows"                  value="$(arg rows)" />
    <param name="cols"                  value="$(arg cols)" />
    <param name="cell_width"            value="$(arg cell_width)" />
    <param name="cell_height"           value="$(arg cell_height)" />

    <param name="sensor_0/name"         value="/$(arg sensor_0_name)" />
    <param name="sensor_0/type"         value="pinhole_rgb" />
    <remap from="~sensor_0/image"       to="/$(arg sensor_0_name)/rgb_rect/image" />
    <remap from="~sensor_0/camera_info" to="/$(arg sensor_0_name)/rgb_rect/camera_info" />

    <param name="sensor_1/name"         value="/$(arg sensor_1_name)" />
    <param name="sensor_1/type"         value="pinhole_rgb" />
    <remap from="~sensor_1/image"       to="/$(arg sensor_1_name)/rgb_rect/image" />
    <remap from="~sensor_1/camera_info" to="/$(arg sensor_1_name)/rgb_rect/camera_info" />

  </node>

</launch>
